pgjdbc/src/main/java/com/yugabyte/ysql/TopologyAwareLoadBalancer.java

```diff
@@ -134,6 +134,109 @@ protected void updateCurrentHostList(ArrayList<String> currentPrivateIps, String
     }
   }
 
+  @Override
+  public boolean hasMorePreferredNode(String chosenHost) {
+    if (hostToPriorityMap.containsKey(chosenHost)) {
+      Object val = hostToPriorityMap.get(chosenHost);
```

Comment:
`hostToPriorityMap.get(chosenHost)` would return `null` if the chosenHost is not in the map, this was raising a NullPointerException since `null `was getting assigned to int variable.

Reply:
Yes, it can return `null` but you don't need to type cast it to Integer explicitly. Suggested code:
```
    if (hostToPriorityMap.containsKey(chosenHost)) {
      Integer chosenHostPriority = hostToPriorityMap.get(chosenHost);
      if (chosenHostPriority != null) {
        for (int i = 1; i < chosenHostPriority; i++) {
        ...
```

Reply:
nit: Just check for non emptyness of hostToPriorityMap also.